### ISSUE TITLE
ServeUnix group argument always 'docker'?

### ISSUE THREAD

**Issue by ahmetalpbalkan** [docker/go-plugins-helpers#38]
It appears like when we create plugin sockets the group must be either `root` or `docker`. Root worked fine for me for a while until I tried my plugin on CoreOS (it basically said `root` group not found, but it was there). So I switched to `docker` group, which I believe the right thing to do.(?)

Is there a use case where this group argument might be something other than `"docker"` value?

**Comment by CWSpear** [comment]
@ahmetalpbalkan why did you close this? What did you discovered? I'm running into the same issue where `root` doesn't work on CoreOS.

**Comment by ahmetalpbalkan** [comment]
I closed due to inactivity. Reopening in case you want to follow up.

**Comment by CWSpear** [comment]
Yeah, indeed, I'm seeing issues with CoreOS giving an error `root group not found`, but it works if you use `docker`. `docker` seems to work elsewhere. I did a search on GitHub and the vast majority seem to pass `root` to `handler.ServeUnix`.

**Comment by dave-tucker** [comment]
I'm pretty sure the `gid` needs to be the same as is used for docker. As for what that `gid` is, it will vary depending on how Docker is packaged. The most sensible option for plugin authors is to allow the `gid` to be user specified, but I think in most cases `root` seems to do the trick.
